src/dc_receive_imf.rs

```diff
@@ -2691,6 +2815,289 @@ mod tests {
+
+        let msg2 = get_chat_msg(&t, chat_id, 1, 2).await;
+        let contact2 = Contact::load_from_db(&t.ctx, msg2.from_id).await.unwrap();
+        assert_eq!(contact2.get_addr(), "notifications@github.com");
+
+        let pseudo_contact = Contact::grpid_to_mailinglist_contact(
+            &t.ctx,
+            "",
+            "deltachat-core-rust.deltachat.github.com",
+            chat_id,
+        )
+        .await
+        .unwrap();
+        assert_eq!(
+            pseudo_contact.get_display_name(),
+            "deltachat/deltachat-core-rust"
+        );
+        assert_eq!(
+            pseudo_contact
+                .param
+                .get(Param::MailingListPseudoContact)
+                .unwrap(),
+            &chat_id.to_u32().to_string()
+        );
+        assert_eq!(msg1.get_override_sender_name().unwrap(), "Max Mustermann");
+        assert_eq!(msg2.get_override_sender_name().unwrap(), "Github");
+    }
+
+    static DC_MAILINGLIST: &[u8] = b"Received: (Postfix, from userid 1000); Mon, 4 Dec 2006 14:51:39 +0100 (CET)\n\
+    From: Bob <bob@posteo.org>\n\
+    To: delta-dev@codespeak.net\n\
+    Subject: Re: [delta-dev] What's up?\n\
+    Message-ID: <38942@posteo.org>\n\
+    List-ID: \"discussions about and around https://delta.chat developments\" <delta.codespeak.net>\n\
+    Precedence: list\n\
+    Date: Sun, 22 Mar 2020 22:37:57 +0000\n\
+    \n\
+    body\n";
+
+    static DC_MAILINGLIST2: &[u8] = b"Received: (Postfix, from userid 1000); Mon, 4 Dec 2006 14:51:39 +0100 (CET)\n\
+    From: Charlie <charlie@posteo.org>\n\
+    To: delta-dev@codespeak.net\n\
+    Subject: Re: [delta-dev] DC is nice!\n\
+    Message-ID: <38943@posteo.org>\n\
+    List-ID: \"discussions about and around https://delta.chat developments\" <delta.codespeak.net>\n\
+    Precedence: list\n\
+    Date: Sun, 22 Mar 2020 22:37:57 +0000\n\
+    \n\
+    body 4\n";
+
+    #[async_std::test]
+    async fn test_classic_mailing_list() {
+        let t = TestContext::new_alice().await;
+        t.ctx
+            .set_config(Config::ShowEmails, Some("2"))
+            .await
+            .unwrap();
+        dc_receive_imf(&t.ctx, DC_MAILINGLIST, "INBOX", 1, false)
+            .await
+            .unwrap();
+        let chats = Chatlist::try_load(&t.ctx, 0, None, None).await.unwrap();
+        let chat_id = chat::create_by_msg_id(&t.ctx, chats.get_msg_id(0).unwrap())
+            .await
+            .unwrap();
+        let chat = Chat::load_from_db(&t.ctx, chat_id).await.unwrap();
+        assert_eq!(chat.name, "delta-dev");
+
+        let msg = get_chat_msg(&t, chat_id, 0, 1).await;
+        let contact1 = Contact::load_from_db(&t.ctx, msg.from_id).await.unwrap();
+        assert_eq!(contact1.get_addr(), "bob@posteo.org");
+        assert_eq!(contact1.param.get(Param::MailingListPseudoContact), None);
+    }
+
+    #[async_std::test]
+    async fn test_mailing_list_decide_block() {
```

Comment:
Better name `test_mailing_list_decide_never`, for consistency (it's yes/never/not now).

Reply:
since merged #2185, options are StartChat/Block/NotNow again, so the name of the test is perfectly fine :)